Snowflake.Data/Core/Session/SFSessionProperty.cs

```diff
@@ -296,6 +296,19 @@ private static void ValidateAuthenticator(SFSessionProperties properties)
                     logger.Error(error);
                     throw new SnowflakeDbException(SFError.UNKNOWN_AUTHENTICATOR, authenticator);
                 }
+
+                if (authenticator.Equals(ExternalBrowserAuthenticator.AUTH_NAME))

```

Comment:
I think this logic would fit better to SFSessionHttpClientProperties class.
In SFSessionHttpClientProperties we interpret string values as proper types values and we do a logic of changing the values.
SFSessionProperty is operates only on string values rather without interpreting them.